include/ofi_hmem.h

```diff
@@ -103,6 +103,7 @@ struct ofi_hmem_ops {
 	int (*host_register)(void *ptr, size_t size);
 	int (*host_unregister)(void *ptr);
 	int (*get_base_addr)(const void *ptr, void **base);
+	bool (*is_ipc_enabled)(void);
```

Comment:
`is_ipc_enabled` is used in this PR to signify that IPC can be performed using exclusively the IPC functions defined in the HMEM interface. My understanding is that Ze needs extra logic on top of that interface, so I did not think it made sense to have `is_ipc_enabled` return true in any case for it. Am I wrong in that regard?

Reply:
Ze needs extra logic in the sense that the device fds need to be exchanged via a socket instead of directly through the shared memory but it still uses the get/open/close handle functions. The p2p check just checks whether the devices can perform an IPC call across the devices. It should be fairly similar to what the CUDA and ROCR versions are checking.

Reply:
Yes, they will all have some check for IPC capability, but my intention was for this function to be about checking if the interface can use this "generic IPC" protocol, i.e. IPC can be done without any interface specific codepaths. If interface specific code is needed, then the function should return false and leave it up to the interface to add the proper conditionals in shm protocol selection. I can make this return true for Ze, but then we are just making protocol selection evaluate more conditionals to get the same effect, especially if more HMEM interfaces need their own special codepath. 
Maybe I am looking at this wrong if CUDA is the odd one out in that it doesn't need extra logic for IPC. If that is the  maybe the HMEM interface should be modified to support the typical use case?

Reply:
I think my problem with this is there's not really a "generic IPC" protocol. It will inherently depend on the underlying interface. CUDA does it one way. ZE does it another way.
Ze needs some interface specific things to make it work which shm does check for and take into account as you suggest but it still uses the "generic" IPC get/open/close handle calls.
All you're doing in the CUDA check for IPC is allocating memory and getting a pointer so by that definition, the same check in ZE would pass. There is nothing in the CUDA IPC check that ZE would fail in the ZE world so I don't think that's enough to say ZE should fail.
Since CUDA was the first major device API to come out it's easy to say "well this API doesn't match up exactly with CUDA so they are the odd ones out" but I think they're just different and require different things. As far as your function that asks "does this interface support IPC?" I think ZE is ok to pass that and have a separate ZE-specific check in shm to see if shm is doing the special extra things to support ZE IPC.

Reply:
That all sounds reasonable to me. I will make these changes now.